src/main/java/com/group6/harmoniq/controllers/SpotifyController.java

```diff
@@ -295,4 +322,188 @@ private Track getTopTrack(String accessToken) throws Exception {
             throw new Exception("Failed to get top tracks", e);
         }
     }
+
+    private List<Artist> getTopArtists(String accessToken) throws Exception {
+
+        // Modify the URL to request top artists (you can adjust the 'limit' parameter)
+        String url = "https://api.spotify.com/v1/me/top/artists?limit=10"; 
+    
+        HttpHeaders headers = new HttpHeaders();
+        headers.set("Authorization", "Bearer " + accessToken);
+    
+        HttpEntity<String> requestEntity = new HttpEntity<>(headers);
+    
+        try {
+            RestTemplate restTemplate = new RestTemplate();
+            ResponseEntity<Map> response = restTemplate.exchange(url, HttpMethod.GET, requestEntity, Map.class);
+    
+            // Extract artist data directly from the response
+            List<Map<String, Object>> topArtistsJson = (List<Map<String, Object>>) response.getBody().get("items");
+            if (topArtistsJson == null) {
+                System.err.println("Error: No 'items' found in Spotify response for top artists.");
+                return Collections.emptyList(); 
+            }
+            
+            // Map the JSON data to Artist objects
+            List<Artist> topArtists = topArtistsJson.stream()
+                .map(artistJson -> {
+                    Artist artist = new Artist();
+                    artist.setName((String) artistJson.get("name"));
+                    
+                    // Extract external URLs
+                    var externalUrls = (Map<String, Object>) artistJson.get("external_urls");
+                    artist.setSpotifyUrl((String) externalUrls.get("spotify"));
+    
+                    // Extract image URL (if available)
+                    var images = (List<Map<String, Object>>) artistJson.get("images");
+                    if (images != null && images.size() > 0) {
+                        artist.setImageUrl((String) images.get(0).get("url"));
+                    }
+    
+                    return artist;
+                })
+                .toList(); 
+    
+            return topArtists;
+        } catch (HttpClientErrorException e) {
+            throw new Exception("Failed to get top artists", e);
+        }
+    }
+
+    private List<Track> getTopTracks(String accessToken) throws Exception {
```

Comment:
Similarly, we can refactor the `getTopTrack` to get the top 10 tracks and avoid the duplicate code